Classes/Domain/NodeCreation/PropertiesHandler.php

```diff
@@ -0,0 +1,176 @@
+                        throw new InvalidReferenceException(sprintf(
+                            'Node with identifier "%s" does not exist.',
+                            $nodeAggregateIdentifier->__toString()
+                        ), 1687632330292);
+                    }
+                    $validReferences[$referenceName] = $node;
+                    continue;
+                }
+
+                if ($referenceType->isReferences()) {
+                    $nodeAggregateIdentifiers = $referenceType->toNodeAggregateIds($referenceValue);
+                    if (count($nodeAggregateIdentifiers) === 0) {
+                        continue;
+                    }
+                    $nodes = [];
+                    foreach ($nodeAggregateIdentifiers as $nodeAggregateIdentifier) {
+                        if (!($nodes[] = $this->subgraph->getNodeByIdentifier($nodeAggregateIdentifier->__toString())) instanceof NodeInterface) {
+                            throw new InvalidReferenceException(sprintf(
+                                'Node with identifier "%s" does not exist.',
+                                $nodeAggregateIdentifier->__toString()
+                            ), 1687632330292);
+                        }
+                    }
+                    $validReferences[$referenceName] = $nodes;
+                    continue;
+                }
+            } catch (InvalidReferenceException $runtimeException) {
+                $caughtExceptions->add(
+                    CaughtException::fromException($runtimeException)
+                        ->withOrigin(sprintf('Reference "%s" in NodeType "%s"', $referenceName, $nodeType->getName()))
+                );
+                continue;
+            }
+        }
+        return $validReferences;
+    }
+
+    /**
+     * In the old CR, it was common practice to set internal or meta properties via this syntax: `_hidden` but we don't allow this anymore.
+     * @throws PropertyIgnoredException
+     */
+    private function assertValidPropertyName($propertyName): void
+    {
+        $legacyInternalProperties = ['_accessRoles', '_contentObject', '_hidden', '_hiddenAfterDateTime', '_hiddenBeforeDateTime', '_hiddenInIndex',
+            '_index', '_name', '_nodeType', '_removed', '_workspace'];
+        if (!is_string($propertyName) || $propertyName === '') {
+            throw new PropertyIgnoredException(sprintf('Because property name must be a non empty string. Got "%s".', $propertyName), 1686149518395);
+        }
+        if ($propertyName[0] === '_') {
+            $lowerPropertyName = strtolower($propertyName);
+            foreach ($legacyInternalProperties as $legacyInternalProperty) {
+                if ($lowerPropertyName === strtolower($legacyInternalProperty)) {
+                    throw new PropertyIgnoredException(sprintf('Because internal legacy property "%s" not implement.', $propertyName), 1686149513158);
```

Comment:
This exception message is weird

Reply:
Yes i agree. I do this in other places too. idk. lets get rid of them see https://github.com/Flowpack/Flowpack.NodeTemplates/issues/new